gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/ContentsUtil.java

```diff
@@ -16,65 +16,63 @@
 
 package com.palantir.gradle.versions.intellij;
 
-import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.progress.ProcessCanceledException;
-import com.intellij.openapi.progress.ProgressIndicator;
-import com.intellij.openapi.progress.ProgressManager;
 import java.io.BufferedReader;
 import java.io.InputStreamReader;
 import java.net.ConnectException;
 import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
-import java.util.Optional;
 import java.util.concurrent.Callable;
-import java.util.concurrent.Future;
 import java.util.stream.Collectors;
+import org.immutables.value.Value;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public final class ContentsUtil {
     private static final Logger log = LoggerFactory.getLogger(ContentsUtil.class);
 
-    public static Optional<String> fetchPageContents(URL pageUrl) {
-        ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
-
-        if (indicator == null) {
-            return Optional.empty();
+    public static ContentResults fetchPageContents(String urlString) {
+        try {
+            URL url = new URL(urlString);
+            return ContentsUtil.fetchPageContents(url);
+        } catch (MalformedURLException e) {
+            log.error("Malformed URL", e);
+            return ContentResults.empty();
```

Comment:
I'd throw here, if we're ever making invalid URLs something has gone wrong in our code